mercury/models/src/main/scala/org/hyperledger/identus/mercury/model/error/package.scala

```diff
@@ -1,11 +1,35 @@
 package org.hyperledger.identus.mercury.model
 
+import org.hyperledger.identus.shared.models._
+
+import java.io.IOException
+
 package object error {
```

Comment:
Do we really need all this complex error types hierarchy? It seems @CryptoKnightIOG was able to simplify it nicely in his last PR, but it was rolled back in this one. To be honest, this `MercuryException|Throwable|Error` construct is something I can't really wrap my head around for some time.

Reply:
hum this is weird because I also remove this. 
I did after making `MercuryError extends Failure`

It was most certainly merge conflict.

@bvoiturier That MercuryThrowable was a hackys solution to integration with the rest of the Open Enterprise Agent. That required Throwable in many places. So instead of adding Throwable in a few more places and making the problem even bigger.
I add the type alias `MercuryThrowable` for me to just know where I need to remove in the future. I even have a comment `// TODO REMOVE Throwable`